src/main/java/org/bladerunnerjs/contrib/contentplugin/appcache/AppcacheManifestBuilder.java

```diff
@@ -52,16 +45,11 @@ public AppcacheManifestBuilder(BRJS brjs, BundleSet bundleSet, String brjsVersio
 	 */
 	public String getManifest(RequestMode requestMode) throws PropertiesException, ContentProcessingException, ConfigException, MalformedTokenException
 	{
-		String manifest = getManifestHeader();
-        if(version != null)
```

Comment:
We definitely had issues around restarting bladerunner and enabling /
disabling appcache in dev. Andy Lyons will know the exact details, could
you check with him please?

On 12 May 2015 at 15:31, Andrew Lyons <notifications@github.com> wrote:

> In
> src/main/java/org/bladerunnerjs/contrib/contentplugin/appcache/AppcacheManifestBuilder.java
> <https://github.com/BladeRunnerJS/brjs-appcache/pull/2#discussion_r30139333>
> :
>
> > @@ -52,16 +45,11 @@ public AppcacheManifestBuilder(BRJS brjs, BundleSet bundleSet, String brjsVersio
> >  	 */
> >  	public String getManifest(RequestMode requestMode) throws PropertiesException, ContentProcessingException, ConfigException, MalformedTokenException
> >  	{
> > -		String manifest = getManifestHeader();
> > -        if(version != null)
>
> I'm not sure this 'if' guard should have been removed; the need for it
> probably wasn't clear (or I've misunderstood something about the refactor).
> Without it, there are annoying delays in dev when you disable appcache
> before you see the uncached files. This is due to the way appcache works:
>
>    - ./brjs serve -v 1.2.3
>    - Load page, and get served
>    - dev.appcache contents:
>
>     # v 1.2.3
>     files...
>
>
>    -
>
>    Appcache caches all the files...
>    -
>
>    Stop brjs
>    - ./brjs serve
>    - Reload page, and get served from appcache
>    - dev.appcache contents:
>
>     # v
>     files...
>
>
>    - Appcache tries to recache all the files, even though we're disabling
>    appcache ...
>    - This takes time. If you refresh the page before all the files have
>    recached, you still get the cached version.
>
> With the 'if' guard, when you reload the page dev.appcache just contains
> '#v' and no files. This causes the appcache to refresh instantly without
> trying to recache all the files, and you see the uncached files as soon as
> you reload.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/BladeRunnerJS/brjs-appcache/pull/2/files#r30139333>.
>



-- 
*Stephen South*
Professional Services Development Manager | Caplin Systems Ltd
Office: +44 (0)20 7826 9618
Mobile: +44 (0)7833 240 370

www.caplin.com
<https://mail.caplin.com/owa/redir.aspx?C=ycOKrX3n8UOuU1SUZqv1hZTBVvrohNEI3NDsCJnyTotUoUMYoptmHmtnEajb7p8dfsEKr963Mak.&URL=http%3a%2f%2fwww.caplin.com%2f>

Single-Dealer Platform News Feed: *SerenDiPity*
<https://mail.caplin.com/owa/redir.aspx?C=ycOKrX3n8UOuU1SUZqv1hZTBVvrohNEI3NDsCJnyTotUoUMYoptmHmtnEajb7p8dfsEKr963Mak.&URL=http%3a%2f%2fsingledealerplatforms.wordpress.com%2f>
Caplin Tech Blog: *Platformability
<https://mail.caplin.com/owa/redir.aspx?C=ycOKrX3n8UOuU1SUZqv1hZTBVvrohNEI3NDsCJnyTotUoUMYoptmHmtnEajb7p8dfsEKr963Mak.&URL=http%3a%2f%2fblog.caplin.com%2f>*